ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/TransactionComponents.kt

```diff
@@ -0,0 +1,196 @@
+package net.corda.v5.ledger.utxo
+
+import net.corda.v5.base.annotations.CordaSerializable
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.crypto.SecureHash
+import net.corda.v5.ledger.common.transaction.Party
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import java.security.PublicKey
+import java.time.Duration
+import java.time.Instant
+import java.util.jar.JarInputStream
+
+/**
+ * Specifies UTXO transaction component groups.
+ *
+ * @property TRANSACTION_PARAMETERS The transaction parameters component group. Ordinal = 0.
+ * @property MEMBERSHIP_PARAMETERS The membership parameters component group. Ordinal = 1.
+ * @property PACKAGES The packages component group. Ordinal = 2.
+ * @property SIGNATORIES The signatories component group. Ordinal = 3.
+ * @property ATTACHMENTS The attachments component group. Ordinal = 4.
+ * @property NOTARY The notary component group. Ordinal = 5.
+ * @property TIME_WINDOW The time window component group. Ordinal = 6.
+ * @property COMMANDS The commands component group. Ordinal = 7.
+ * @property COMMANDS_METADATA The commands metadata component group. Ordinal = 8.
+ * @property INPUTS The inputs component group. Ordinal = 9.
+ * @property INPUTS_METADATA The inputs metadata component group. Ordinal = 10.
+ * @property OUTPUTS The outputs component group. Ordinal = 11.
+ * @property OUTPUTS_METADATA The outputs metadata component group. Ordinal = 12.
+ * @property REFERENCES The references component group. Ordinal = 13.
+ * @property REFERENCES_METADATA The references metadata component group. Ordinal = 14.
+ */
+enum class ComponentGroup {
+    TRANSACTION_PARAMETERS, MEMBERSHIP_PARAMETERS, PACKAGES, SIGNATORIES, ATTACHMENTS, NOTARY, TIME_WINDOW, COMMANDS, COMMANDS_METADATA, INPUTS, INPUTS_METADATA, OUTPUTS, OUTPUTS_METADATA, REFERENCES, REFERENCES_METADATA
+}
+
+/**
+ * Defines a data attachment that can be referenced within a transaction.
+ *
+ * Attachments are ZIP, or optionally a signed JAR file that contains one or more files of public, static data that
+ * can be referenced from a transaction, and utilised within a contract. Examples of attachments include:
+ *
+ * - Calendar data
+ * - Fixes (e.g. LIBOR)
+ * - Smart contract code
+ * - Legal documents
+ * - Facts generated by oracles.
+ *
+ * @property id The unique hash of the attachment.
+ * @property size The size of the attachment in bytes.
+ * @property signatories The keys that have been used to sign the attachment, or empty if the attachment does not require signing.
+ */
+@CordaSerializable
+@DoNotImplement
+interface Attachment {
+
+    val id: SecureHash
+    val size: Int
+    val signatories: Set<PublicKey>
```

Comment:
Totally forgot about that corner of the API! Will have to remember it when calculating required signers as my mind just assumes set union of keys on commands is sufficient!